bcl2fastq/index_summary.py

```diff
@@ -297,5 +211,49 @@ def update_pie_chart(run_alias, known_json, unknown_json):
     )
 
 
+@app.callback(
+    [dep.Output('run_select', 'value'),
+     dep.Output('error', 'displayed')],
+    [dep.Input('url', 'pathname')]
+)
+def change_url(value):
+    """ Allows user to enter Run name in URL which will update dropdown automatically, and the graphs.
+        If User enters any value that's not a valid run an error box will pop up and return user to most recent run
+
+        Parameters:
+             value: the URL input after the port (the pathname).
+
+        Returns:
+            The string value (without '/') of the user input for the drop-down to use
+            Error pop-up displayed depending on user input.
+    """
+    #   In a pathname, it automatically adds '/' to the beginning of the input even if pathname blank
+    #   While page loads, pathname is set to 'None'. Once page is loaded pathname is set to user input.
+    if value == "/" or value is None:
+        return all_runs[0], False
+    elif value[1:] not in all_runs:
+        return all_runs[0], True
+    else:
+        return value[1:], False
+
+
+@app.callback(
+    [dep.Output('known_index_bar', 'figure'),
+     dep.Output('unknown_index_bar', 'figure'),
+     dep.Output('known_unknown_pie', 'figure'),
+     dep.Output('known_fraction', 'value')],
+    [dep.Input('run_select', 'value')]
+
+)
+def update_layout(run_alias):
+    """ When input(run dropdown) is changed, known index bar, unknown index bar, piechart and textarea are called
+        Returns:
+            functions update_known_index_bar, update_unknown_index_bar, update_pie_chart's data value,
+            and update_pie_chart's fraction value
+    """
+    return update_known_index_bar(run_alias), update_unknown_index_bar(run_alias), update_pie_chart(run_alias)[0], \
```

Comment:
Instead of calling `update_pie_chart` twice to get the two values, you could call it once and use multiple variable assignment to save the resulting values, and then `return` those.